fedimint-server/src/consensus/engine.rs

```diff
@@ -472,6 +484,81 @@ impl ConsensusEngine {
             .expect("This is the only place where we write to this key");
     }
 
+    /// Creates the directory within the data directory for storing the database
+    /// checkpoints.
+    fn create_checkpoint_directory(&self) -> anyhow::Result<()> {
+        let checkpoint_dir = self.data_dir.join("db_checkpoints");
+
+        if !checkpoint_dir.exists() {
+            fs::create_dir_all(&checkpoint_dir)?;
+        }
+
+        Ok(())
+    }
+
+    /// Creates a backup of the database in the checkpoint directory. The
+    /// frequency of the checkpoints is controlled by
+    /// `FM_DB_CHECKPOINT_FREQ_ENV`. These checkpoints can be used to
+    /// restore the database in case the federation falls out of consensus.
+    fn checkpoint_database(&self, session_index: u64) {
+        // By default, checkpoint the database every 10 sessions
+        let checkpoint_frequency: u64 = env::var(FM_DB_CHECKPOINT_FREQ_ENV)
```

Comment:
Moving reading this to a new function, and in some previous performance-sensitive code, I used `Once` to make sure it's as fast as possible.

Reply:
Moved to `run`